### DISCUSSION TITLE
How to make task3 dependent on Snowflake procedure output of task 1 and 2?

### DISCUSSION THREAD

**Discussion by kjn70** [apache/airflow#30998] [General]
I'm new to Airflow and I am trying to make a conditional workflow that is dependent on the procedural return value of a Snowflake function.

task 1 and 2 run a Snowflake procedure and will either do something or return "No Change" in the return value.  Then task 3, if either of the values is something other than "No Change" should run.  Otherwise it should not run.

In this test dag, currently task 1 and 2 run in parallel and then task 3 runs.  Currently the procedure simply outputs what you put into it for testing (real-world it will not a lot of other stuff..)

`from asyncore import file_dispatcher
from datetime import datetime, timedelta
from pickle import TUPLE2
from textwrap import dedent
from airflow import DAG
from airflow.operators.python import PythonOperator
from airflow.providers.snowflake.operators.snowflake import SnowflakeOperator
from airflow.operators.email_operator import EmailOperator
from airflow.models import Variable
from airflow.utils.task_group import TaskGroup

<redacted>

with DAG(
    ret_test',
    default_args={
        'depends_on_past': False,
        'wait_for_downstream': False,
    },
    description='Test for capturing return value from procedure',
    start_date=datetime(2023, 4, 1),
    catchup=False,
) as dag:

    t1 = SnowflakeOperator(
        task_id = 'Test1',
        sql = 'call PRC$RET_TEST(\'No Change\');'
    )
    
    t2 = SnowflakeOperator(
        task_id = 'Test2',
        sql = 'call PRC$RET_TEST(\'No Change\');'
    )

    t3 = SnowflakeOperator(
        task_id = 'rebuild_CAL_MSTR',
        sql = 'call PRC$CAL_UPD_MSTR(\'Hello World!\');'
    )

    t1
    t2
    [t1, t2] >> t3
`

Some parts of the dag were redacted but it gives the general idea.  

Any thoughts on how to capture the return value and build an expression that determines how task 3 runs?  I had thought this would be easy but I'm currently at a loss.

**Comment by pankajastro** [comment]
I think you can use some branching func https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#branching
